extpreimage/invoice.go

```diff
@@ -9,11 +9,18 @@ import (
 	"github.com/lightningnetwork/lnd/lnwire"
 )
 
-// Invoice is a simplified view of a channeldb.Invoice that includes
-// the information needed to lookup an external preimage. We do
-// this to move as much logic out out of the main LND packages as
-// possible.
-type Invoice struct {
+// Invoice is an interface for using external and local preimages
+// with channeldb.Invoices. We define it here as an interface
+// to move as much logic out of the main LND packages as possible.
+type Invoice interface {
+	GetPaymentHash() ([32]byte, error)
+	GetPaymentPreimage(uint32, uint32, Client, InvoiceRegistry) (
+		[32]byte, error, error)
+}
+
+// CastInvoice implements the Invoice interface and is in a format
+// to allow channeldb.Invoice to be cast into it.
+type CastInvoice struct {
```

Comment:
@dannypaz @martineehrlich better ideas for the name of this struct? I need it to be public, but I don't want to conflict with the interface `Invoice` that I've created.